cmake/Platform/Linux/DetachDebugSymbols.cmake

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+
+if(NOT ${CMAKE_ARGC} EQUAL 6)
+    message(FATAL_ERROR "DetachDebugSymbols script called with the wrong number of arguments: ${CMAKE_ARGC}")
+endif()
+
+string(TOLOWER ${CMAKE_ARGV4} STATIC_LIBRARY_SUFFIX)
+string(TOLOWER ${CMAKE_ARGV5} DEBUG_SYMBOL_EXT)
+
+# Make sure that both gnu tools 'strip' and 'objcopy' are available
+find_program(GNU_STRIP_TOOL strip)
+find_program(GNU_OBJCOPY objcopy)
+
+if (GNU_STRIP_TOOL AND GNU_OBJCOPY)
+
+    # Only perform the debug symbol detachment on non-static libraries. (You cannot add the gnu debug link on an static library)
+    get_filename_component(file_extension ${CMAKE_ARGV3} LAST_EXT)
+    if (NOT ${file_extension} STREQUAL ${STATIC_LIBRARY_SUFFIX})
```

Comment:
This check is flaky to me.
We should pass in the TARGET name into this function and then use get_property to read the library TYPE from the target. We should compare that library TYPE against the value of STATIC_LIBRARY(Here is an example of doing that: https://github.com/o3de/o3de/blob/development/cmake/Platform/Common/Install_common.cmake#L224)

Since executables on Linux don't have an extension, it is acceptable to have an executable with a name of `executable.a`.
Even if the naming scheme is non-sensical

Reply:
Actually, I can remove the check. The target linking type is determined before this call and we can remove the custom command call to this for static libs